### PR TITLE
Added "Remove" link for apps, removes app from storage and FxOS

### LINKED ISSUE #45: "Remove App" for app dashboard
Removes an app from B2G profile and the add-on storage.

### PR DESCRIPTION
**Pull request by digitarald**
Fixes #45

### CODE REVIEW

File: addon/lib/main.js [hunk 1]
```diff
@@ -217,6 +222,46 @@ let simulator = {
...
+    delete webapps[config.xkey];
+    File.open(webappsFile, "w").writeAsync(
+      JSON.stringify(webapps, null, 2) + "\n",
+      function(error) {
+        if (error) {
+          console.error("Error writing webapp record to registry: " + error);
+          return;
+        }
+
+        // Create target folder
```
Comment (mykmelez) [inline addon/lib/main.js]: Nit: this comment should describe _removing_ a folder.

File: addon/data/content/js/main.js [hunk 2]
```diff
@@ -121,12 +127,21 @@ var Simulator = {
               container.append(
                 $("<div class='app'>").append(
                   $("<div class='options'>").append(
+                    $("<a href='#'>")
+                      .addClass("button")
+                      .text("Remove")
+                      .click(function(evt) {
+                        evt.preventDefault();
+                        if (confirm("This will remove the app from Firefox OS Simulator and the Manager. Are you sure?")) {
```
Comment (mykmelez) [inline addon/data/content/js/main.js]: I suppose this is ok, although ideally we'd remove without prompting while giving the user a way to undo the action (as does Add-ons Manager when you remove an addon).

Comment (mykmelez) [inline addon/data/content/js/main.js]: Nit: this line should be shortened to 80 characters or less in width.

Comment (digitarald) [inline addon/data/content/js/main.js]: Agreed, the add-on Undo feature seems easier for the user. Storage would add a `deleted` flag and flush deleted addons from storage at unload. Should I try to get this in v1?

Comment (mykmelez) [inline addon/data/content/js/main.js]: If it isn't hard to do, then yes! If it's going to take longer than the first half of today, though, then no. In that case, let's land this and do the additional work for the next cycle.

**Pull request merged**